ui/src/utils/lineWriter.ts

```diff
@@ -0,0 +1,237 @@
+import AJAX from 'src/utils/ajax'
+
+export interface Tags {
+  [key: string]: string
+}
+
+export interface Fields {
+  [key: string]: number | string
+}
+
+export enum Precision {
+  ns = 'ns',
+  u = 'u',
+  ms = 'ms',
+  s = 's',
+  m = 'm',
+  h = 'h',
+}
+
+export interface BatchOptions {
+  maxIntervalInSeconds: number
+  maxBatchedLines: number
+}
+
+const defaultBatchOptions: BatchOptions = {
+  maxIntervalInSeconds: 10,
+  maxBatchedLines: 100,
+}
+
+const nowInSeconds = function nowInSeconds() {
+  return Math.floor(Date.now() / 1000)
+}
+
+export class LineWriter {
+  protected url: string
+  protected orgId: number | string
+  protected bucketName: string
+  protected authToken: string
+  protected batchOptions: BatchOptions
+
+  protected timerID: number | null
+  protected batchedLines: number
+  protected lines: string[]
+
+  constructor(
+    url: string,
+    orgId: number | string,
```

Comment:
```suggestion
    orgID: number | string,
```
nit here (i prefer `Id`) actually.  sadly, the rest of our app uses `ID`